shared/strbuf.c

```diff
@@ -62,8 +66,6 @@ char *strbuf_steal(struct strbuf *buf)
 
 const char *strbuf_str(struct strbuf *buf)
 {
-	if (!buf_grow(buf, buf->used + 1))
-		return NULL;
```

Comment:
That makes something like:
```
struct strbuf buf;
strbuf_init(&buf);
if (condition)
    strbuf_pushchars(&buf, "test");
const char *s = strbuf_str(&buf);
```
problematic rather than strbuf_str just returning NULL when condition is false.